File: build-zip.py
import zipfile
import os

dist_dir = r'g:\LTD\fc-dist'
zip_path = r'g:\LTD\fc-deploy.zip'

skip_dirs = {'.cache', '.pnpm', '__pycache__'}

with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zf:
    for root, dirs, files in os.walk(dist_dir):
        # Skip unwanted directories
        dirs[:] = [d for d in dirs if d not in skip_dirs]
        
        for f in files:
            full_path = os.path.join(root, f)
            # Calculate relative path from dist_dir
            rel_path = os.path.relpath(full_path, dist_dir)
            # Convert to forward slashes for Linux compatibility
            zip_path_entry = rel_path.replace(os.sep, '/')
            zf.write(full_path, zip_path_entry)
            print(f"  Added: {zip_path_entry}")

size_mb = os.path.getsize(zip_path) / (1024 * 1024)
print(f"\nDone! {zip_path} ({size_mb:.2f} MB)")

File: build-zip.js
const fs = require('fs');
const path = require('path');
const { createWriteStream } = require('fs');
const archiver = require('archiver');

// 如果没装 archiver，用内置 zip 方式
const AdmZip = require('adm-zip');

const zip = new AdmZip();
const distDir = path.join(__dirname, 'fc-dist');

function addDir(dir, prefix) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    const zipPath = prefix ? `${prefix}/${entry.name}` : entry.name;
    if (entry.isDirectory()) {
      if (entry.name === '.cache' || entry.name === '.pnpm') continue;
      addDir(fullPath, zipPath);
    } else {
      zip.addLocalFile(fullPath, prefix || '');
    }
  }
}

addDir(distDir, '');
zip.writeZip(path.join(__dirname, 'fc-deploy.zip'));
console.log('Zip created with adm-zip');
